src/server/transport.rs

```diff
@@ -86,29 +88,42 @@ pub trait RaftStoreRouter: Send + Clone {
 pub struct ServerRaftStoreRouter {
     pub ch: SendCh<StoreMsg>,
     pub significant_msg_sender: Sender<SignificantMsg>,
+    local_reader_ch: Scheduler<ReadTask>,
 }
 
 impl ServerRaftStoreRouter {
     pub fn new(
-        ch: SendCh<StoreMsg>,
+        raftstore_ch: SendCh<StoreMsg>,
         significant_msg_sender: Sender<SignificantMsg>,
+        local_reader_ch: Scheduler<ReadTask>,
     ) -> ServerRaftStoreRouter {
         ServerRaftStoreRouter {
-            ch,
+            ch: raftstore_ch,
             significant_msg_sender,
+            local_reader_ch,
         }
     }
 }
 
 impl RaftStoreRouter for ServerRaftStoreRouter {
     fn try_send(&self, msg: StoreMsg) -> RaftStoreResult<()> {
-        self.ch.try_send(msg)?;
-        Ok(())
+        if ReadTask::acceptable(&msg) {
```

Comment:
How about provide different method for different messages?

Reply:
It'a good idea, I will send another PR, since this PR is large enough.